docs/rules/class-methods-use-this.md

```diff
@@ -0,0 +1,85 @@
+# Enforce that class methods utilize `this` (class-methods-use-this)
+
+If a class method does not use `this`, it can safely be made a static function.
+
+It's possible to have a class method which doesn't use `this`, such as:
+
+```js
+class A {
+    constructor() {
+        this.a = "hi";
+    }
+
+    print() {
+        console.log(this.a);
+    }
+
+    sayHi() {
+        console.log("hi");
+    }
+}
+
+let aObj = new A();
+a.sayHi(); // => "hi"
+```
+
+In the example above, the `sayHi` method doesn't use `this`, so we can make it a static method:
+
+```js
+class A {
+    constructor() {
+        this.a = "hi";
+    }
+
+    print() {
+        console.log(this.a);
+    }
+
+    static sayHi() {
+        console.log("hi");
+    }
+}
+
+A.sayHi(); // => "hi"
+```
+
+## Rule Details
+
+This rule is aimed to flag class methods that do not use `this`.
+
+Examples of **incorrect** code for this rule:
+
+```js
+/*eslint class-methods-use-this: "error"*/
+/*eslint-env es6*/
+
+class A {
+    foo() {
+        console.log("Hello World");     /*error Expected 'this' to be used by class method 'foo'.*/
+    }
+}
+```
+
+Examples of **correct** code for this rule:
+
+```js
+/*eslint class-methods-use-this: "error"*/
+/*eslint-env es6*/
+class A {
+    foo() {
+        this.bar = "Hello World"; // OK, this is used
+    }
+}
+
+class A {
+    constructor() {
+        super(); // OK. constructor is exempt
```

Comment:
Sorry, this is invalid syntax.
If a class doesn't have `extends` clause, SuperCall throws a SyntaxError.
Also, this function is a correct code whether this is a constructor or not, since this has a `super`.